cmd/agent/app/reporter/tchannel/reporter.go

```diff
@@ -34,19 +34,19 @@ const (
 
 type batchMetrics struct {
 	// Number of successful batch submissions to collector
-	BatchesSubmitted metrics.Counter `metric:"batches.submitted"`
+	BatchesSubmitted metrics.Counter `metric:"tc-reporter.batches.submitted"`
```

Comment:
1. there is no need to add tc-reporter explicitly like that, since you can pass that string as the first argument to Namespace in L76 (see point 3 below).
2. since the metrics are changing anyway, maybe we should rename them to use underscores, as we did with other backend metrics?
3. I think historically `tc` was used to refer to `tcollector`, which was the TChannel name of the collector service (now it's `jaeger-collector`). When we implement #927, there will be a different type of reporter, e.g. `grpc-reporter`. Also, in some cases we internally run the agent with two reporters (using OtherReporters configuration of the builder), I don't even know if they distinguish their metrics. So my suggestion, at minimum, to rename this to `tchannel-reporter`, to make it more obvious what we're talking about, but we also need to figure out how metrics will work after #927. 

cc @black-adder

Reply:
> 2. since the metrics are changing anyway, maybe we should rename them to use underscores, as we did with other backend metrics?

I can change it using underscore like `tchannel_reporter`. Just curious, the reason I kept using `-` is normalizer in jaeger-lib already replaces them. Don't we need it?
https://github.com/jaegertracing/jaeger-lib/blob/master/metrics/prometheus/factory.go#L109

Reply:
actually I see that collector metrics are still using dots as separators, so never mind about understores